packages/client/src/ActionsAndReducers/playerUi/playerUi_Reducer.js

```diff
@@ -95,34 +95,49 @@ export const playerUiReducer = (state = initialState, action) => {
     return { isParticipant, allRolesIncluded, observing, templates }
   }
 
-  const modifyPlatformTypesBasedOnMessages = (platformTypes, newState) => {
-    // TODO: modyfy platform types based on laydown or other mapControll type messages
-    console.log(platformTypes)
-    return platformTypes
+  const modifyForcesBasedOnMessages = ({ allPlatformTypes, channels, allForces }) => {
+    let res = allForces
+    const mapChannel = Object.values(channels).find(({ name }) => (name === 'Mapping'))
+    if (mapChannel && mapChannel.messages && mapChannel.messages.length) {
+      for (const message of mapChannel.messages) {
+        res = modifyForcesBasedOnMessage(allForces, message)
+      }
+    }
+    return res
   }
 
-  const handleForceLaydown = (/* object */ payload, /* object */ newState) => {
+  const modifyForcesBasedOnMessage = (allForces, message) => {
+    let res = allForces
+    if (message.details && message.details.forceDelta) {
+      res = handleForceDelta(message, allForces)
+    }
+    return res
+  }
+
+  const handleForceLaydown = (/* object */ message, /* object */ allForces) => {
     // find the force
-    const force = newState.allForces.find(item => item.name === payload.force)
```

Comment:
@lilitkarapetyan - you switched to using `findIndex` rather than `find`. Was this to solve a particular problem? Or is it just more robust programming method?